src/pocketmine/entity/ThrownPotion.php

```diff
@@ -86,11 +88,10 @@ public function onUpdate($currentTick){
 			$this->kill();
 			$this->close();
 			$hasUpdate = true;
-		}
-		
-		if($this->onGround) {
+		} elseif($this->onGround) {
 			$this->kill();
```

Comment:
Wasn't looking to mess with the logic, just trying to fix the bug. I'll do some tests and see if that condition is needed.